art/estimators/object_detection/pytorch_yolo.py

```diff
@@ -397,50 +423,161 @@ def predict(self, x: np.ndarray, batch_size: int = 128, **kwargs) -> List[Dict[s
+                       and providing it takes no effect.
+        """
+        import torch
+        import torchvision
+
+        # Set model to train mode
+        self._model.train()
+
+        if self._optimizer is None:  # pragma: no cover
+            raise ValueError("An optimizer is needed to train the model, but none for provided.")
+
+        # Apply preprocessing
+        x_preprocessed, y_preprocessed = self._apply_preprocessing(x, y=y, fit=False, no_grad=True)
+
+        # Convert samples into tensors
+        transform = torchvision.transforms.Compose([torchvision.transforms.ToTensor()])
+
+        if self.clip_values is not None:
+            norm_factor = self.clip_values[1]
+        else:
+            norm_factor = 1.0
+
+        if self.channels_first:
+            x_preprocessed = torch.from_numpy(x_preprocessed / norm_factor).to(self.device)
+        else:
+            x_preprocessed = torch.stack([transform(x_i / norm_factor).to(self.device) for x_i in x_preprocessed])
+
+        # Convert labels into tensors, if needed
+        if isinstance(y_preprocessed[0]["boxes"], np.ndarray):
+            y_preprocessed_tensor = []
+            for y_i in y_preprocessed:
+                y_preprocessed_t = {
+                    "boxes": torch.from_numpy(y_i["boxes"]).type(torch.float).to(self.device),
+                    "labels": torch.from_numpy(y_i["labels"]).type(torch.int64).to(self.device),
+                }
+                if "masks" in y_i:
+                    y_preprocessed_t["masks"] = torch.from_numpy(y_i["masks"]).type(torch.uint8).to(self.device)
+                y_preprocessed_tensor.append(y_preprocessed_t)
+            y_preprocessed = y_preprocessed_tensor
+        y_preprocessed = np.asarray(y_preprocessed)
+
+        num_batch = len(x_preprocessed) / float(batch_size)
+        if drop_last:
+            num_batch = int(np.floor(num_batch))
+        else:
+            num_batch = int(np.ceil(num_batch))
+        ind = np.arange(len(x_preprocessed))
+
+        # Start training
+        for _ in range(nb_epochs):
+            # Shuffle the examples
+            np.random.shuffle(ind)
+
+            # Train for one epoch
+            for m in range(num_batch):
+                i_batch = x_preprocessed[ind[m * batch_size : (m + 1) * batch_size]]
+                o_batch = y_preprocessed[ind[m * batch_size : (m + 1) * batch_size]]
+
+                # Zero the parameter gradients
+                self._optimizer.zero_grad()
+
+                # Form the loss function
+                labels_t = translate_labels_art_to_yolov3(labels_art=o_batch)
+                loss_components = self._model(i_batch, labels_t)
+                if isinstance(loss_components, dict):
+                    loss = sum(loss_components.values())
+                else:
+                    loss = loss_components
+
+                # Do training
+                loss.backward()  # type: ignore
```

Comment:
Yes, this is necessary since the `sum(...)` function is type annotated to return an `int` type. This is not actually true since for a list of tensors, it returns a tensor. However, this is what mypy believes which causes the typing error.

Reply:
Ok, thank you.